ParseCAN/spec/bus/frame.py

```diff
@@ -97,18 +99,40 @@ def __post_init__(self):
         self.atom.extend(atom)
 
     def unpack(self, frame, **kwargs):
-        return {atom.name: atom.unpack(frame, **kwargs) for atom in self.atom}
+        return (self.name,
+                {atom.name: atom.unpack(frame, **kwargs)
```

Comment:
It is true that you can't have an `Atom` as a key yet, because it is mutable. I think we still need to have the tuple convention be that the actual object comes first and not the object name. The reason is that you can use the type information to figure out where exactly you are.
Let's do something like `(self, [(atom, atom.unpack(frame, **kwargs)) for atom in self.atom])` and when and if we get immutability we can make this an ordered dict.